include/logging/log_backend.h

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright (c) 2018 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#ifndef LOG_BACKEND_H
+#define LOG_BACKEND_H
+
+#include <logging/log_msg.h>
+#include <assert.h>
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Forward declaration of the log_backend type. */
+struct log_backend;
+
+/**
+ * @brief Logger backend API.
+ */
+struct log_backend_api {
+	void (*put)(struct log_backend const *const backend,
+		    struct log_msg *msg);
+
+	void (*panic)(struct log_backend const *const backend);
+};
+
+/**
+ * @brief Logger backend control block.
+ */
+struct log_backend_cb {
```

Comment:
Zephyr codebase in many places uses abbreviation "cb" for "callback". It will be quite confusing to have it mean "control block" here, so please avoid that.

Reply:
renamed to log_backend_control_block